internal/controller/managedcluster_controller.go

```diff
@@ -297,13 +295,21 @@ func (r *ManagedClusterReconciler) updateCluster(ctx context.Context, mc *hmc.Ma
 		return ctrl.Result{}, nil
 	}
 
-	helmValues, err := setIdentityHelmValues(mc.Spec.Config, cred.Spec.IdentityRef)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error setting identity values: %w", err)
+	if err := mc.AddHelmValues(func(values map[string]any) error {
+		values["clusterIdentity"] = cred.Spec.IdentityRef
+
+		if _, ok := values["clusterLabels"]; !ok {
+			// Use the ManagedCluster's own labels if not defined.
+			values["clusterLabels"] = mc.GetObjectMeta().GetLabels()
```

Comment:
It doesn't seem reliable since there's no guarantee that all cluster templates have this field (unless you want to make this field required).

Reply:
Yes, in this approach we are depending on the helm chart for cluster templates to use the `clusterLabels` value. We do need some way to assign labels to `Cluster` objects. Maybe the approach would be?
1. In the HMC controller, use the values defined in `clusterLabels` in the ManagedCluster spec and patch `Cluster` to apply those labels to it.
2. Or if `clusterLabels` is not defined, then use the ManagedCluster's own labels.